### ISSUE TITLE
[cmake] after #2574 set frugally-deep_FOUND to FALSE so package "frugally-deep" is considered to be NOT FOUND

### ISSUE THREAD

**Issue by junliume** [ROCmSoftwarePlatform/MIOpen#2607]
**Update**: Issue reproduced at newer releases, investigating if this is a more common issue than just backward incompatibility

This breaks compatibility with ROCm 5.7.0.

The build in my 5.7.0 docker (rocm/miopen:ci_429410) fails with the following:
```bash
CMake Error at src/CMakeLists.txt:736 (find_package):
  Found package configuration file:

    /usr/local/lib/cmake/frugally-deep/frugally-deepConfig.cmake

  but it set frugally-deep_FOUND to FALSE so package "frugally-deep" is
  considered to be NOT FOUND.

-- Configuring incomplete, errors occurred!
```
I had to `-DMIOPEN_ENABLE_AI_KERNEL_TUNING=Off -DMIOPEN_ENABLE_AI_IMMED_MODE_FALLBACK=Off` to W/A this.

With 6.0 RC (rocm/miopen:ci_fbea27) and 5.7.1 (rocm/miopen:ci_3eb168) dockers, everything is fine.

_Originally posted by @atamazov in https://github.com/ROCm/MIOpen/issues/2574#issuecomment-1856896416_

**Comment by junliume** [comment]
To provide more info, in the failing docker above, the error is because `frugally-deep_GLOBAL_FOUND` is FALSE while `frugally-deep_FIND_REQUIRED_GLOBAL` is TRUE.
I do not know why it worked in one docker but not the other though.

**Comment by junliume** [comment]
@atamazov @JehandadKhan @apwojcik Unfortunately I have to raise this ticket to blocker level, because the recommended way of building MIOpen seems to be broken:
```
export LC_ALL=C.UTF-8 && \
export LANG=C.UTF-8 && \
apt-get update -y && \
apt-get install -y sudo && \
sudo apt-get -y install git git-lfs unzip lbzip2 python3.9-venv pkg-config vim ninja-build half && \
sudo apt update && \
cd /root && git clone -b develop https://github.com/ROCmSoftwarePlatform/MIOpen && cd MIOpen && \
mkdir -p install_dir && cmake -P install_deps.cmake --minimum --prefix /root/MIOpen/install_dir && \
mkdir -p build && \
cd build && \
CXX=/opt/rocm/llvm/bin/clang++ CC=/opt/rocm/llvm/bin/clang cmake -DMIOPEN_BACKEND=HIP -DCMAKE_PREFIX_PATH="/root/MIOpen/install_dir;/opt/rocm" \
-DMIOPEN_TEST_ALL=On -DMIOPEN_TEST_HALF=On .. && \
make -j$(nproc) install
```

this file:
```
{install_dir}/lib/cmake/frugally-deep/frugally-deepConfig.cmake
```
where
```
macro(check_required_components _NAME)
  foreach(comp ${${_NAME}_FIND_COMPONENTS})
    if(NOT ${_NAME}_${comp}_FOUND)
      if(${_NAME}_FIND_REQUIRED_${comp})
        	message(FATAL_ERROR "${_NAME}_${comp}") <=== added this line to debug
	      set(${_NAME}_FOUND FALSE)
      endif()
    endif()
  endforeach()
endmacro()
```
shows that `frugally-deep_GLOBAL` is not found but required

**Comment by apwojcik** [comment]
The issue is not caused by the ROCm 5.7.0 but by the CMake version installed in the docker. The command `find_package(... GLOBAL)` requires CMake version >= 3.24

**Comment by atamazov** [comment]
@apwojcik @junliume So the fix is changing `cmake_minimum_required( VERSION 3.15 )` to `cmake_minimum_required( VERSION 3.24 )`

**Comment by apwojcik** [comment]
That is the part of the fix. This way, we let everyone know that the CMake version needs to be higher. The other part is to upgrade CMake to 3.24 in the docker. I have provided #2610, which should solve the issue without upgrading CMake.

**Comment by atamazov** [comment]
@apwojcik But if everything works with #2610, then let's stay with it and close the ticket, right?

UPD: and #2608 is not necessary anymore, right?

Thanks for quick feedback))

**Comment by apwojcik** [comment]
Correct. If #2610 is working, we don't need #2608, and we don't need a CMake update for now.

**Comment by junliume** [comment]
Fixed with #2610
